rlog/DRAFT-fiat-shamir.mdx

```diff
@@ -0,0 +1,183 @@
+Thus, $c$ fixes the group elements $g$, $X$ and $T$.
+
+We note that the elements that would be omitted in the hash by applying weak Fiat-Shamir heuristic are $g$ and $X$.
+
+### Chaum-Pedersen protocol
+A tuple of group elements $(U,V,W)$ is a DH-triple if and only if there exists some $x \in \mathbb{Z}_p$ so that $V = g^x$ and $W = U^x$.
+The Chaum-Pedersen protocol provides a framework that enables a Prover to convince a Verifier that she possesses such a $x$ for a claimed DH-triple $(U,V,W)$.
+The Chaum-Pedersen protocol proceeds as follows:
+
+|Prover| | Verifier|
+|----|----|----|
+|$t \in_R \mathbb{Z}_p$, $T := g^t$, $S := U^t$| $\stackrel{T,S}{\longrightarrow}$| |
+|  | $\stackrel{c}{\longleftarrow}$ | $c \in_R \mathbb{Z}_p$|
+| $z := t + xc$ | $\stackrel{z}{\longrightarrow}$ | |
+| | | output 1 provided $g^z \stackrel{?}{=} T V^c$ and $U^z \stackrel{?}{=} SW^c$|
+
+The message $c = Hash(g||U||V||W||T||S)$ when the Prover applies the strong Fiat-Shamir heuristic.
+The properties of $\mathsf{Hash}$ fixes the generator $g$ and the Prover's statement $(U,V,W)$.
+
+
+## Improper use of the Fiat-Shamir heuristic
+
+The Fiat-Shamir heuristic appears to be a fairly straight-forward technique to implement.
+However, a subtle but serious issue that can occur in the application of the Fiat-Shamir heuristic has been a point of discussion for the past few years.
+The issue concerns what messages are included in the hash.
+In particular, are the public parameters used to compute the hash value?
+
+Bernhard et al. ([11](https://eprint.iacr.org/2016/771.pdf)) provide a discussion of Fiat-Shamir heuristic resticted to $\Sigma$-protocols.
+In particular, Bernhard et al. discuss the pitfalls of the weak Fiat-Shamir heuristic.
+Recall that the strong Fiat-Shamir heuristic requires that the public parameters are included in the calculations of the Verifier's messages while the weak version does not.
+The inclusion of the public parameters in the hash evaluations fix these public values for the entire protocol.
+This means that the Prover cannot retroactively change them.
+
+The issues with the differences in the variants of the Fiat-Shamir heuristics has persisted since Bernhard et al.'s paper. 
+In recent years, auditors from [Trail of Bits](https://www.trailofbits.com/) and [OpenZepellin](https://www.openzeppelin.com/) have 
+released blogs ([2](https://blog.trailofbits.com/2022/04/13/part-1-coordinated-disclosure-of-vulnerabilities-affecting-girault-bulletproofs-and-plonk/),
+ [3](https://blog.trailofbits.com/2022/04/14/the-frozen-heart-vulnerability-in-giraults-proof-of-knowledge/),
+ [4](https://blog.trailofbits.com/2022/04/15/the-frozen-heart-vulnerability-in-bulletproofs/), [5](https://blog.trailofbits.com/2022/04/18/the-frozen-heart-vulnerability-in-plonk/), [7](https://blog.openzeppelin.com/the-last-challenge-attack))
+ and papers ([6](https://eprint.iacr.org/2023/691), [8](https://eprint.iacr.org/2024/398)) 
```

Comment:
Can we add this [paper](https://eprint.iacr.org/2023/1071.pdf) who examines the security of different fiat-shamir on zkSNARKs into the papers.

Reply:
Added.